tests/test_qnode.py

```diff
@@ -922,6 +922,135 @@ def circuit(a, b):
                                       [0.] * 2 + [0.] * 5 + [-np.sin(b[2, 1])]])  # expval 2
         assert np.allclose(circuit_jacobian, expected_jacobian, atol=tol, rtol=0)
 
+    def test_controlled_RX_gradient(self, tol):
+        """Test gradient of controlled RX gate"""
+        dev = qml.device('default.qubit', wires=2)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.PauliX(wires=0)
+            qml.CRX(x, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        a = 0.572   #gradient is zero for all `a` for this particular circuit
+
+	# get the analytic gradient
+        gradA = circuit.jacobian([a], method='A')
+	# get the finite difference gradient
+        gradF = circuit.jacobian([a], method='F')
+
+	# the expected gradient
+        expected = 0
+
+        assert np.allclose(gradF, expected, atol=tol, rtol=0)
+        assert np.allclose(gradA, expected, atol=tol, rtol=0)
+
+        @qml.qnode(dev)
+        def circuit1(x):
+            qml.PauliX(wires=0)
+            qml.PauliX(wires=1)
+            qml.CRX(x, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        b = 0.123 #gradient is zero for all `b` for this particular circuit
```

Comment:
(applies throughout): Would be better to have a circuit which give non-zero gradient (since circuit above already captures a case of zero gradient).